File: src/schemas/school_admin.py
from datetime import date, datetime
from decimal import Decimal
from typing import Any

from pydantic import BaseModel, ConfigDict, Field


# Certificate Template Schemas
class CertificateTemplateBase(BaseModel):
    template_name: str = Field(..., max_length=200)
    certificate_type: str = Field(..., max_length=50)
    template_config: dict[str, Any]
    is_default: bool = False
    is_active: bool = True


class CertificateTemplateCreate(CertificateTemplateBase):
    institution_id: int


class CertificateTemplateUpdate(BaseModel):
    template_name: str | None = Field(None, max_length=200)
    certificate_type: str | None = Field(None, max_length=50)
    template_config: dict[str, Any] | None = None
    is_default: bool | None = None
    is_active: bool | None = None


class CertificateTemplateResponse(CertificateTemplateBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    created_at: datetime
    updated_at: datetime


# Issued Certificate Schemas
class GenerateTransferCertificateRequest(BaseModel):
    student_id: int
    leaving_date: date
    reason_for_leaving: str
    conduct: str = "Good"
    remarks: str | None = None
    template_id: int | None = None


class GenerateLeavingCertificateRequest(BaseModel):
    student_id: int
    leaving_date: date
    reason: str
    conduct: str = "Good"
    remarks: str | None = None
    template_id: int | None = None


class GenerateBonafideCertificateRequest(BaseModel):
    student_id: int
    academic_year: str
    purpose: str
    remarks: str | None = None
    template_id: int | None = None


class GenerateCharacterCertificateRequest(BaseModel):
    student_id: int
    conduct_assessment: str
    behavioral_remarks: str
    remarks: str | None = None
    template_id: int | None = None


class GenerateStudyCertificateRequest(BaseModel):
    student_id: int
    from_date: date
    to_date: date
    purpose: str
    remarks: str | None = None
    template_id: int | None = None


class GenerateConductCertificateRequest(BaseModel):
    student_id: int
    conduct_rating: str
    remarks: str | None = None
    template_id: int | None = None


class GenerateMigrationCertificateRequest(BaseModel):
    student_id: int
    migration_to: str
    migration_reason: str
    last_exam_passed: str | None = None
    remarks: str | None = None
    template_id: int | None = None


class GenerateFeeCertificateRequest(BaseModel):
    student_id: int
    academic_year: str
    total_fees_paid: float
    remarks: str | None = None
    template_id: int | None = None


class GenerateNoDuesCertificateRequest(BaseModel):
    student_id: int
    remarks: str | None = None
    template_id: int | None = None


class IssueCertificateRequest(BaseModel):
    certificate_type: str
    data: dict[str, Any]
    template_id: int | None = None
    remarks: str | None = None


class IssuedCertificateResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    student_id: int
    certificate_type: str
    template_id: int | None
    serial_number: str
    issue_date: date
    data_snapshot: dict[str, Any]
    pdf_url: str | None
    issued_by: int | None
    remarks: str | None
    status: str
    created_at: datetime
    updated_at: datetime


class RevokeCertificateRequest(BaseModel):
    reason: str | None = None


# ID Card Template Schemas
class IDCardTemplateBase(BaseModel):
    template_name: str = Field(..., max_length=200)
    layout_config: dict[str, Any]
    color_scheme: str | None = Field(None, max_length=100)
    logo_position: str | None = Field(None, max_length=50)
    fields_to_show: dict[str, Any] | None = None
    orientation: str = "portrait"
    is_default: bool = False


class IDCardTemplateCreate(IDCardTemplateBase):
    institution_id: int


class IDCardTemplateUpdate(BaseModel):
    template_name: str | None = Field(None, max_length=200)
    layout_config: dict[str, Any] | None = None
    color_scheme: str | None = Field(None, max_length=100)
    logo_position: str | None = Field(None, max_length=50)
    fields_to_show: dict[str, Any] | None = None
    orientation: str | None = None
    is_default: bool | None = None


class IDCardTemplateResponse(IDCardTemplateBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    created_at: datetime
    updated_at: datetime


class GenerateIDCardRequest(BaseModel):
    student_id: int
    template_id: int | None = None
    valid_until: date | None = None


class GenerateBulkIDCardsRequest(BaseModel):
    section_id: int | None = None
    grade_id: int | None = None
    template_id: int | None = None
    valid_until: date | None = None


# Staff Member Schemas
class StaffMemberBase(BaseModel):
    employee_id: str = Field(..., max_length=50)
    first_name: str = Field(..., max_length=100)
    last_name: str = Field(..., max_length=100)
    email: str | None = Field(None, max_length=255)
    phone: str | None = Field(None, max_length=20)
    designation: str | None = Field(None, max_length=200)
    department: str = Field(..., max_length=50)
    date_of_birth: date | None = None
    gender: str | None = Field(None, max_length=20)
    address: str | None = None
    qualification: str | None = Field(None, max_length=255)
    joining_date: date | None = None
    bank_account_number: str | None = Field(None, max_length=50)
    bank_name: str | None = Field(None, max_length=200)
    ifsc_code: str | None = Field(None, max_length=20)
    pan_number: str | None = Field(None, max_length=20)
    aadhar_number: str | None = Field(None, max_length=20)
    basic_salary: Decimal | None = None
    photo_url: str | None = Field(None, max_length=500)


class StaffMemberCreate(StaffMemberBase):
    institution_id: int


class StaffMemberUpdate(BaseModel):
    employee_id: str | None = Field(None, max_length=50)
    first_name: str | None = Field(None, max_length=100)
    last_name: str | None = Field(None, max_length=100)
    email: str | None = Field(None, max_length=255)
    phone: str | None = Field(None, max_length=20)
    designation: str | None = Field(None, max_length=200)
    department: str | None = Field(None, max_length=50)
    date_of_birth: date | None = None
    gender: str | None = Field(None, max_length=20)
    address: str | None = None
    qualification: str | None = Field(None, max_length=255)
    joining_date: date | None = None
    leaving_date: date | None = None
    bank_account_number: str | None = Field(None, max_length=50)
    bank_name: str | None = Field(None, max_length=200)
    ifsc_code: str | None = Field(None, max_length=20)
    pan_number: str | None = Field(None, max_length=20)
    aadhar_number: str | None = Field(None, max_length=20)
    basic_salary: Decimal | None = None
    photo_url: str | None = Field(None, max_length=500)
    is_active: bool | None = None
    status: str | None = None


class StaffMemberResponse(StaffMemberBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    leaving_date: date | None
    is_active: bool
    status: str
    created_at: datetime
    updated_at: datetime


# Staff Payroll Schemas
class StaffPayrollBase(BaseModel):
    staff_id: int
    month: int = Field(..., ge=1, le=12)
    year: int
    basic_salary: Decimal
    hra: Decimal = Decimal('0.00')
    da: Decimal = Decimal('0.00')
    ta: Decimal = Decimal('0.00')
    special_allowance: Decimal = Decimal('0.00')
    pf_deduction: Decimal = Decimal('0.00')
    esi_deduction: Decimal = Decimal('0.00')
    professional_tax: Decimal = Decimal('0.00')
    tds: Decimal = Decimal('0.00')
    other_deductions: Decimal = Decimal('0.00')
    gross_salary: Decimal
    net_salary: Decimal


class StaffPayrollCreate(StaffPayrollBase):
    institution_id: int


class StaffPayrollUpdate(BaseModel):
    basic_salary: Decimal | None = None
    hra: Decimal | None = None
    da: Decimal | None = None
    ta: Decimal | None = None
    special_allowance: Decimal | None = None
    pf_deduction: Decimal | None = None
    esi_deduction: Decimal | None = None
    professional_tax: Decimal | None = None
    tds: Decimal | None = None
    other_deductions: Decimal | None = None
    gross_salary: Decimal | None = None
    net_salary: Decimal | None = None
    payment_status: str | None = None
    payment_date: date | None = None
    transaction_reference: str | None = None


class StaffPayrollResponse(StaffPayrollBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    payment_status: str
    payment_date: date | None
    transaction_reference: str | None
    created_at: datetime
    updated_at: datetime


# SMS Template Schemas
class SMSTemplateBase(BaseModel):
    template_name: str = Field(..., max_length=200)
    template_type: str = Field(..., max_length=50)
    message_body: str
    is_active: bool = True


class SMSTemplateCreate(SMSTemplateBase):
    institution_id: int


class SMSTemplateUpdate(BaseModel):
    template_name: str | None = Field(None, max_length=200)
    template_type: str | None = Field(None, max_length=50)
    message_body: str | None = None
    is_active: bool | None = None


class SMSTemplateResponse(SMSTemplateBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    created_at: datetime
    updated_at: datetime


# Enquiry Record Schemas
class EnquiryRecordBase(BaseModel):
    enquiry_date: date
    student_name: str = Field(..., max_length=200)
    parent_name: str = Field(..., max_length=200)
    parent_phone: str = Field(..., max_length=20)
    parent_email: str | None = Field(None, max_length=255)
    enquiry_for_grade: str | None = Field(None, max_length=50)
    source: str = Field(..., max_length=50)
    notes: str | None = None
    assigned_to: int | None = None
    follow_up_date: date | None = None


class EnquiryRecordCreate(EnquiryRecordBase):
    institution_id: int


class EnquiryRecordUpdate(BaseModel):
    enquiry_date: date | None = None
    student_name: str | None = Field(None, max_length=200)
    parent_name: str | None = Field(None, max_length=200)
    parent_phone: str | None = Field(None, max_length=20)
    parent_email: str | None = Field(None, max_length=255)
    enquiry_for_grade: str | None = Field(None, max_length=50)
    source: str | None = Field(None, max_length=50)
    status: str | None = None
    notes: str | None = None
    assigned_to: int | None = None
    follow_up_date: date | None = None


class EnquiryRecordResponse(EnquiryRecordBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    institution_id: int
    status: str
    sms_sent_count: int
    last_sms_sent_at: datetime | None
    created_at: datetime
    updated_at: datetime


# Bulk Operations
class BulkStaffImportRow(BaseModel):
    employee_id: str
    first_name: str
    last_name: str
    email: str | None = None
    phone: str | None = None
    designation: str | None = None
    department: str
    date_of_birth: str | None = None
    gender: str | None = None
    qualification: str | None = None
    joining_date: str | None = None
    basic_salary: str | None = None


class StaffFilterParams(BaseModel):
    department: str | None = None
    status: str | None = None
    is_active: bool | None = None
    search: str | None = None


class EnquiryFilterParams(BaseModel):
    source: str | None = None
    status: str | None = None
    assigned_to: int | None = None
    from_date: date | None = None
    to_date: date | None = None
    search: str | None = None


class SendSMSRequest(BaseModel):
    template_id: int
    recipient_type: str  # 'student', 'parent', 'enquiry'
    recipient_ids: list[int]
    variables: dict[str, str] | None = None


class PayrollGenerateRequest(BaseModel):
    month: int = Field(..., ge=1, le=12)
    year: int
    staff_ids: list[int] | None = None


class CertificateStatistics(BaseModel):
    total_issued: int
    issued_by_type: dict[str, int]
    recent_certificates: list[IssuedCertificateResponse]


class StaffStatistics(BaseModel):
    total_staff: int
    active_staff: int
    inactive_staff: int
    staff_by_department: dict[str, int]
    staff_by_status: dict[str, int]


class EnquiryStatistics(BaseModel):
    total_enquiries: int
    enquiries_by_status: dict[str, int]
    enquiries_by_source: dict[str, int]
    conversion_rate: float

File: frontend/src/api/schoolAdmin.ts
import axios from '@/lib/axios';

// Certificate Types
export type CertificateType =
  | 'TC'
  | 'LC'
  | 'Bonafide'
  | 'Character'
  | 'Study'
  | 'Conduct'
  | 'Migration'
  | 'Fee'
  | 'No Dues'
  | 'Sports'
  | 'Merit'
  | 'Participation';

export interface Certificate {
  id: number;
  institution_id: number;
  student_id: number;
  student_name?: string;
  certificate_type: CertificateType;
  serial_number: string;
  issue_date: string;
  template_id?: number;
  remarks?: string;
  pdf_url?: string;
  issued_by_id: number;
  issued_by_name?: string;
  is_revoked: boolean;
  revoked_at?: string;
  revoked_by?: number;
  revoked_reason?: string;
  created_at: string;
  updated_at: string;
}

export interface IssueCertificateRequest {
  student_id: number;
  certificate_type: CertificateType;
  template_id?: number;
  remarks?: string;
  issue_date?: string;
}

export interface CertificateListParams {
  skip?: number;
  limit?: number;
  certificate_type?: CertificateType;
  student_id?: number;
  from_date?: string;
  to_date?: string;
  is_revoked?: boolean;
  search?: string;
}

export interface CertificateListResponse {
  items: Certificate[];
  total: number;
  skip: number;
  limit: number;
}

export interface RevokeCertificateRequest {
  reason: string;
}

// ID Card Types
export interface IDCardTemplate {
  id: number;
  institution_id: number;
  name: string;
  orientation: 'portrait' | 'landscape';
  front_config: IDCardFaceConfig;
  back_config: IDCardFaceConfig;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface IDCardFaceConfig {
  background_color?: string;
  header_color?: string;
  border_color?: string;
  logo_url?: string;
  show_photo?: boolean;
  show_name?: boolean;
  show_admission_number?: boolean;
  show_class?: boolean;
  show_dob?: boolean;
  show_blood_group?: boolean;
  show_address?: boolean;
  show_phone?: boolean;
  show_parent_phone?: boolean;
  show_emergency_contact?: boolean;
  custom_fields?: Array<{ label: string; value: string }>;
}

export interface IDCardTemplateCreate {
  name: string;
  orientation: 'portrait' | 'landscape';
  front_config: IDCardFaceConfig;
  back_config: IDCardFaceConfig;
  is_default?: boolean;
}

export interface IDCardTemplateUpdate {
  name?: string;
  orientation?: 'portrait' | 'landscape';
  front_config?: IDCardFaceConfig;
  back_config?: IDCardFaceConfig;
  is_default?: boolean;
}

export interface BulkIDCardGenerateRequest {
  grade_id?: number;
  section_id?: number;
  template_id?: number;
}

// Certificate Template Types
export interface CertificateTemplate {
  id: number;
  institution_id: number;
  name: string;
  certificate_type: CertificateType;
  template_config: CertificateTemplateConfig;
  is_default: boolean;
  created_at: string;
  updated_at: string;
}

export interface CertificateTemplateConfig {
  header?: string;
  logo_url?: string;
  signature_url?: string;
  watermark_url?: string;
  body_text?: string;
  footer_text?: string;
  custom_fields?: Array<{ key: string; label: string; placeholder?: string }>;
  styles?: {
    header_font?: string;
    body_font?: string;
    header_color?: string;
    body_color?: string;
    border_color?: string;
  };
}

export interface CertificateTemplateCreate {
  name: string;
  certificate_type: CertificateType;
  template_config: CertificateTemplateConfig;
  is_default?: boolean;
}

export interface CertificateTemplateUpdate {
  name?: string;
  certificate_type?: CertificateType;
  template_config?: CertificateTemplateConfig;
  is_default?: boolean;
}

// Staff Types
export interface Staff {
  id: number;
  institution_id: number;
  employee_id: string;
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  department?: string;
  designation?: string;
  joining_date?: string;
  leaving_date?: string;
  salary?: number;
  bank_name?: string;
  account_number?: string;
  ifsc_code?: string;
  pan_number?: string;
  address?: string;
  emergency_contact?: string;
  qualification?: string;
  experience_years?: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface StaffCreate {
  employee_id: string;
  first_name: string;
  last_name: string;
  email?: string;
  phone?: string;
  department?: string;
  designation?: string;
  joining_date?: string;
  salary?: number;
  bank_name?: string;
  account_number?: string;
  ifsc_code?: string;
  pan_number?: string;
  address?: string;
  emergency_contact?: string;
  qualification?: string;
  experience_years?: number;
  is_active?: boolean;
}

export interface StaffUpdate {
  employee_id?: string;
  first_name?: string;
  last_name?: string;
  email?: string;
  phone?: string;
  department?: string;
  designation?: string;
  joining_date?: string;
  leaving_date?: string;
  salary?: number;
  bank_name?: string;
  account_number?: string;
  ifsc_code?: string;
  pan_number?: string;
  address?: string;
  emergency_contact?: string;
  qualification?: string;
  experience_years?: number;
  is_active?: boolean;
}

export interface StaffListParams {
  skip?: number;
  limit?: number;
  department?: string;
  designation?: string;
  is_active?: boolean;
  search?: string;
}

export interface StaffListResponse {
  items: Staff[];
  total: number;
  skip: number;
  limit: number;
}

// Payroll Types
export interface PayrollRecord {
  id: number;
  institution_id: number;
  staff_id: number;
  staff_name?: string;
  employee_id?: string;
  department?: string;
  month: string;
  year: number;
  basic_salary: number;
  hra?: number;
  da?: number;
  allowances?: number;
  deductions?: number;
  gross_salary: number;
  net_salary: number;
  payment_date?: string;
  payment_status: 'pending' | 'paid' | 'cancelled';
  created_at: string;
  updated_at: string;
}

export interface PayrollRecordCreate {
  staff_id: number;
  month: string;
  year: number;
  basic_salary: number;
  hra?: number;
  da?: number;
  allowances?: number;
  deductions?: number;
  payment_date?: string;
}

export interface PayrollRecordUpdate {
  basic_salary?: number;
  hra?: number;
  da?: number;
  allowances?: number;
  deductions?: number;
  payment_date?: string;
  payment_status?: 'pending' | 'paid' | 'cancelled';
}

export interface PayrollListParams {
  skip?: number;
  limit?: number;
  staff_id?: number;
  month?: string;
  year?: number;
  payment_status?: 'pending' | 'paid' | 'cancelled';
}

export interface PayrollListResponse {
  items: PayrollRecord[];
  total: number;
  skip: number;
  limit: number;
}

export interface GeneratePayrollRequest {
  month: string;
  year: number;
  department?: string;
}

export interface PayrollSummary {
  total_staff: number;
  total_gross: number;
  total_deductions: number;
  total_net: number;
  department_breakdown: Array<{
    department: string;
    staff_count: number;
    total_gross: number;
    total_net: number;
  }>;
}

export interface BulkPayrollUpdateRequest {
  payroll_ids: number[];
  payment_status?: 'pending' | 'paid' | 'cancelled';
  payment_date?: string;
}

// SMS Template Types
export interface SMSTemplate {
  id: number;
  institution_id: number;
  name: string;
  template_type: string;
  message_template: string;
  variables?: string[];
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface SMSTemplateCreate {
  name: string;
  template_type: string;
  message_template: string;
  variables?: string[];
  is_active?: boolean;
  [key: string]: unknown;
}

export interface SMSTemplateUpdate {
  name?: string;
  template_type?: string;
  message_template?: string;
  variables?: string[];
  is_active?: boolean;
}

export interface SendSMSRequest {
  template_id: number;
  recipient_ids: number[];
  recipient_type: 'student' | 'parent' | 'teacher' | 'staff';
  variables?: Record<string, string>;
}

// Enquiry Types
export interface Enquiry {
  id: number;
  institution_id: number;
  student_name: string;
  parent_name: string;
  parent_email?: string;
  parent_phone: string;
  enquiry_date: string;
  grade_interested?: string;
  status: 'new' | 'contacted' | 'visited' | 'follow_up' | 'converted' | 'closed';
  notes?: string;
  follow_up_date?: string | null;
  assigned_to_id?: number;
  assigned_to_name?: string;
  created_at: string;
  updated_at: string;
}

export interface EnquiryCreate {
  student_name: string;
  parent_name: string;
  parent_email?: string;
  parent_phone: string;
  enquiry_date?: string;
  grade_interested?: string;
  notes?: string;
  follow_up_date?: string;
  assigned_to_id?: number;
  [key: string]: unknown;
}

export interface EnquiryUpdate {
  student_name?: string;
  parent_name?: string;
  parent_email?: string;
  parent_phone?: string;
  grade_interested?: string;
  status?: 'new' | 'contacted' | 'visited' | 'follow_up' | 'converted' | 'closed';
  notes?: string;
  follow_up_date?: string | null;
  assigned_to_id?: number;
}

export interface EnquiryListParams {
  skip?: number;
  limit?: number;
  status?: 'new' | 'contacted' | 'visited' | 'follow_up' | 'converted' | 'closed';
  grade_interested?: string;
  from_date?: string;
  to_date?: string;
  search?: string;
}

export interface EnquiryListResponse {
  items: Enquiry[];
  total: number;
  skip: number;
  limit: number;
}

// API Object
const schoolAdminApi = {
  // Certificate APIs
  certificates: {
    list: async (params?: CertificateListParams): Promise<CertificateListResponse> => {
      const response = await axios.get('/api/v1/certificates/', { params });
      return response.data;
    },

    get: async (id: number): Promise<Certificate> => {
      const response = await axios.get(`/api/v1/certificates/${id}`);
      return response.data;
    },

    issue: async (data: IssueCertificateRequest): Promise<Certificate> => {
      const response = await axios.post('/api/v1/certificates/issue', data);
      return response.data;
    },

    download: async (id: number): Promise<Blob> => {
      const response = await axios.get(`/api/v1/certificates/${id}/download`, {
        responseType: 'blob',
      });
      return response.data;
    },

    revoke: async (id: number, data: RevokeCertificateRequest): Promise<Certificate> => {
      const response = await axios.post(`/api/v1/certificates/${id}/revoke`, data);
      return response.data;
    },

    preview: async (data: IssueCertificateRequest): Promise<Blob> => {
      const response = await axios.post('/api/v1/certificates/preview', data, {
        responseType: 'blob',
      });
      return response.data;
    },
  },

  // Certificate Template APIs
  certificateTemplates: {
    list: async (certificateType?: CertificateType): Promise<CertificateTemplate[]> => {
      const response = await axios.get('/api/v1/certificate-templates/', {
        params: certificateType ? { certificate_type: certificateType } : undefined,
      });
      return response.data;
    },

    get: async (id: number): Promise<CertificateTemplate> => {
      const response = await axios.get(`/api/v1/certificate-templates/${id}`);
      return response.data;
    },

    create: async (data: CertificateTemplateCreate): Promise<CertificateTemplate> => {
      const response = await axios.post('/api/v1/certificate-templates/', data);
      return response.data;
    },

    update: async (id: number, data: CertificateTemplateUpdate): Promise<CertificateTemplate> => {
      const response = await axios.put(`/api/v1/certificate-templates/${id}`, data);
      return response.data;
    },

    delete: async (id: number): Promise<void> => {
      await axios.delete(`/api/v1/certificate-templates/${id}`);
    },
  },

  // ID Card APIs
  idCards: {
    generate: async (studentId: number, templateId?: number): Promise<Blob> => {
      const response = await axios.post(
        `/api/v1/id-cards/generate`,
        { student_id: studentId, template_id: templateId },
        { responseType: 'blob' }
      );
      return response.data;
    },

    bulkGenerate: async (data: BulkIDCardGenerateRequest): Promise<Blob> => {
      const response = await axios.post('/api/v1/id-cards/bulk-generate', data, {
        responseType: 'blob',
      });
      return response.data;
    },
  },

  // ID Card Template APIs
  idCardTemplates: {
    list: async (): Promise<IDCardTemplate[]> => {
      const response = await axios.get('/api/v1/id-card-templates/');
      return response.data;
    },

    get: async (id: number): Promise<IDCardTemplate> => {
      const response = await axios.get(`/api/v1/id-card-templates/${id}`);
      return response.data;
    },

    create: async (data: IDCardTemplateCreate): Promise<IDCardTemplate> => {
      const response = await axios.post('/api/v1/id-card-templates/', data);
      return response.data;
    },

    update: async (id: number, data: IDCardTemplateUpdate): Promise<IDCardTemplate> => {
      const response = await axios.put(`/api/v1/id-card-templates/${id}`, data);
      return response.data;
    },

    delete: async (id: number): Promise<void> => {
      await axios.delete(`/api/v1/id-card-templates/${id}`);
    },

    uploadLogo: async (id: number, file: File): Promise<{ logo_url: string }> => {
      const formData = new FormData();
      formData.append('file', file);
      const response = await axios.post(`/api/v1/id-card-templates/${id}/upload-logo`, formData, {
        headers: {
          'Content-Type': 'multipart/form-data',
        },
      });
      return response.data;
    },
  },

  // Staff APIs
  staff: {
    list: async (params?: StaffListParams): Promise<StaffListResponse> => {
      const response = await axios.get('/api/v1/staff/', { params });
      return response.data;
    },

    get: async (id: number): Promise<Staff> => {
      const response = await axios.get(`/api/v1/staff/${id}`);
      return response.data;
    },

    create: async (data: StaffCreate): Promise<Staff> => {
      const response = await axios.post('/api/v1/staff/', data);
      return response.data;
    },

    update: async (id: number, data: StaffUpdate): Promise<Staff> => {
      const response = await axios.put(`/api/v1/staff/${id}`, data);
      return response.data;
    },

    delete: async (id: number): Promise<void> => {
      await axios.delete(`/api/v1/staff/${id}`);
    },

    bulkImport: async (
      file: File
    ): Promise<{ imported: number; failed: number; errors?: string[] }> => {
      const formData = new FormData();
      formData.append('file', file);
      const response = await axios.post('/api/v1/staff/bulk-import', formData, {
        headers: {
          'Content-Type': 'multipart/form-data',
        },
      });
      return response.data;
    },

    downloadTemplate: async (): Promise<Blob> => {
      const response = await axios.get('/api/v1/staff/import-template', {
        responseType: 'blob',
      });
      return response.data;
    },
  },

  // Payroll APIs
  payroll: {
    list: async (params?: PayrollListParams): Promise<PayrollListResponse> => {
      const response = await axios.get('/api/v1/payroll/', { params });
      return response.data;
    },

    get: async (id: number): Promise<PayrollRecord> => {
      const response = await axios.get(`/api/v1/payroll/${id}`);
      return response.data;
    },

    create: async (data: PayrollRecordCreate): Promise<PayrollRecord> => {
      const response = await axios.post('/api/v1/payroll/', data);
      return response.data;
    },

    update: async (id: number, data: PayrollRecordUpdate): Promise<PayrollRecord> => {
      const response = await axios.put(`/api/v1/payroll/${id}`, data);
      return response.data;
    },

    delete: async (id: number): Promise<void> => {
      await axios.delete(`/api/v1/payroll/${id}`);
    },

    generatePayslip: async (id: number): Promise<Blob> => {
      const response = await axios.get(`/api/v1/payroll/${id}/payslip`, {
        responseType: 'blob',
      });
      return response.data;
    },

    generatePayroll: async (
      data: GeneratePayrollRequest
    ): Promise<{ created: number; message: string }> => {
      const response = await axios.post('/api/v1/payroll/generate', data);
      return response.data;
    },

    getSummary: async (month: string, year: number): Promise<PayrollSummary> => {
      const response = await axios.get('/api/v1/payroll/summary', {
        params: { month, year },
      });
      return response.data;
    },

    bulkUpdate: async (data: BulkPayrollUpdateRequest): Promise<{ updated: number }> => {
      const response = await axios.post('/api/v1/payroll/bulk-update', data);
      return response.data;
    },

    exportReport: async (month: string, year: number, format: 'excel' | 'pdf'): Promise<Blob> => {
      const response = await axios.get('/api/v1/payroll/export', {
        params: { month, year, format },
        responseType: 'blob',
      });
      return response.data;
    },
  },

  // SMS Template APIs
  smsTemplates: {
    list: async (): Promise<SMSTemplate[]> => {
      const response = await axios.get('/api/v1/sms-templates/');
      return response.data;
    },

    get: async (id: number): Promise<SMSTemplate> => {
      const response = await axios.get(`/api/v1/sms-templates/${id}`);
      return response.data;
    },

    create: async (data: SMSTemplateCreate): Promise<SMSTemplate> => {
      const response = await axios.post('/api/v1/sms-templates/', data);
      return response.data;
    },

    update: async (id: number, data: SMSTemplateUpdate): Promise<SMSTemplate> => {
      const response = await axios.put(`/api/v1/sms-templates/${id}`, data);
      return response.data;
    },

    delete: async (id: number): Promise<void> => {
      await axios.delete(`/api/v1/sms-templates/${id}`);
    },

    sendSMS: async (data: SendSMSRequest): Promise<{ sent: number; failed: number }> => {
      const response = await axios.post('/api/v1/sms-templates/send', data);
      return response.data;
    },
  },

  // Enquiry APIs
  enquiries: {
    list: async (params?: EnquiryListParams): Promise<EnquiryListResponse> => {
      const response = await axios.get('/api/v1/enquiries/', { params });
      return response.data;
    },

    get: async (id: number): Promise<Enquiry> => {
      const response = await axios.get(`/api/v1/enquiries/${id}`);
      return response.data;
    },

    create: async (data: EnquiryCreate): Promise<Enquiry> => {
      const response = await axios.post('/api/v1/enquiries/', data);
      return response.data;
    },

    update: async (id: number, data: EnquiryUpdate): Promise<Enquiry> => {
      const response = await axios.put(`/api/v1/enquiries/${id}`, data);
      return response.data;
    },

    delete: async (id: number): Promise<void> => {
      await axios.delete(`/api/v1/enquiries/${id}`);
    },
  },
};

export default schoolAdminApi;
